internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -183,6 +183,21 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 			nextCheckTimes = append(nextCheckTimes, nextCheck)
 		}
 	}
+
+	// check certificate expiration
```

Comment:
Let's makes sure this abides to MaxUnhelthy/others MHC API fields

Reply:
I performed a test locally and it does abide to the `MaxUnhealthy`/`UnhealthyRange` API. 

Although, once thing to consider is that in most cases all the replicas of the control plane are probabaly spinned up together and they all will become unhealthy roughtly at the same time. It will be possible that since all the machine become healthy at the same time MHC remediation will be short-circuited because of `MaxUnhealthy`/`UnhealthyRange`, leading to no certificate rotation. 

We should definitely continue to honor the `MaxUnhealthy`/`UnhealthyRange` API, I am just pointing out that the above coult be a common case.

Reply:
I would assume/hope that in the KCP case one Machine after another is remediated, right?